src/ScalarMessaging.js

```diff
@@ -355,12 +390,20 @@ function setWidget(event, roomId) {
             };
         }
 
+        // This starts listening for when the echo comes back from the server
+        // since the widget won't appear added until this happens. If we don't
+        // wait for this, the action will complete but if the user is fast enough,
+        // the widget still won't actually be there.
         client.setAccountData('m.widgets', userWidgets).then(() => {
+            return waitForUserWidget(widgetId);
+        }).then(() => {
             sendResponse(event, {
                 success: true,
             });
 
             dis.dispatch({ action: "user_widget_updated" });
+        }).catch((e) => {
```

Comment:
We should at least log this error or it'll be dropped on the floor. Bonus points for sending it back to scalar.

Reply:
That's what `sendError` does (both).

Reply:
done